packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -245,16 +245,17 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
     };
 
     let onBlur = (e) => {
-      let isInAnyScope = isElementInAnyScope(e.relatedTarget, scopes);
-
-      if (!isInAnyScope) {
-        activeScope = scopeRef;
-        focusedNode.current = e.target;
-        // Firefox doesn't shift focus back to the Dialog properly without this
-        raf.current = requestAnimationFrame(() => {
+      // Firefox doesn't shift focus back to the Dialog properly without this
+      raf.current = requestAnimationFrame(() => {
```

Comment:
interesting, you just needed to move the original one to include more of the logic? how come you don't need to call e.persist?

Reply:
hmm, didn't know about e.persist, but logging out the event inside the raf returns the correct event (doesn't return null which I think is what you are worried about here due to event pooling?) 

I'm not sure why that is from a brief google search...